group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMember.java

```diff
@@ -301,11 +322,9 @@ public String toString() {
     private final int previousMemberEpoch;
 
     /**
-     * The next member epoch. This corresponds to the target
-     * assignment epoch used to compute the current assigned,
-     * revoking and assigning partitions.
+     * The member state.
      */
-    private final int targetMemberEpoch;
+    private final MemberState state;
```

Comment:
so that i understand: a member no longer has a targetMemberEpoch and we rely on the member state to determine which state in the reconciliation process the member is in. is this correct?

i vaguely remember fencing requests based on member epochs but looking at this PR it looks like targetMemberEpoch is only used to determine which state we should transition the member to.

Reply:
Is the idea with the new code that we just want to match the desired state (ie, add the assigned and remove the revoked)

If we change the assignment then we will just bump the epoch and fence older ones?